src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs

```diff
@@ -243,12 +243,13 @@ protected override CompletionItemRules GetCompletionItemRules(ImmutableArray<(IS
             if (symbol.IsKind(SymbolKind.Method))
             {
                 var isInferredTypeDelegate = context.InferredTypes.Any(type => type.IsDelegateType());
-                if (!isInferredTypeDelegate)
+                if (!context.IsNameOfContext && !isInferredTypeDelegate)
                 {
                     item = SymbolCompletionItem.AddShouldProvideParenthesisCompletion(item);
                 }
             }
-            else if (symbol.IsKind(SymbolKind.NamedType) || symbol is IAliasSymbol aliasSymbol && aliasSymbol.Target.IsType)
+            else if (symbol.IsKind(SymbolKind.NamedType)
+                || symbol is IAliasSymbol aliasSymbol && aliasSymbol.Target.IsType)
```

Comment:
this is just all sorts of confusingly precedented.

Reply:
📝 There are no actual changes here aside from a new line, so it would be fine to just merge these back into one line and leave the code unchanged